storage/rocksdb/ha_rocksdb_proto.h

```diff
@@ -90,4 +90,8 @@ Rdb_dict_manager *rdb_get_dict_manager(void)
 class Rdb_ddl_manager;
 Rdb_ddl_manager *rdb_get_ddl_manager(void)
     MY_ATTRIBUTE((__warn_unused_result__));
+
+struct Rdb_hton_init_state;
+Rdb_hton_init_state *rdb_get_hton_init_state(void)
```

Comment:
Why not a reference instead?

Reply:
I considered it and was torn. My initial implementation was a ref only and I changed it. I was trying to stay sort of inline with existing MyRocks semantics. There is a difference though, all of the existing MyRocks constructs that follow this model are dynamic whereas this is pretty much a static singleton. Really I was considering what upstream would be more inclined to take in and trying to fit that.